### PR TITLE
clippy: make generated code nice to read

### PR DESCRIPTION
**Pull request by DesmondWillowbrook**
Feel free to close if this is too minor.

(For context, I _have_ read the clippy policy in `dev/style.md`)

### CODE REVIEW

**Comment by lnicola** [comment]
Thanks. We already use `matches!`, so I guess it's all right.

@bors r+

**Comment by lnicola** [comment]
This should actually fail the tests AFAICT.

**Comment by DesmondWillowbrook** [comment]
Oh right. I'll get to that now.

**Comment by lnicola** [comment]
You can run the failing test and add the modified file.

**Comment by DesmondWillowbrook** [comment]
Done now!

**Comment by lnicola** [comment]
Let's wait a bit and see if the formatting changes bother anyone :sweat_smile:.

**Comment by Veykril** [comment]
> Let's wait a bit and see if the formatting changes bother anyone 😅.

It kind of does 😅 While using `matches!` is usually better, we shouldn't be doing that in this here I think. It's generated, so using a macro when we don't have to is actually worse, in theory it reduces compile times, although in this case its obviously negible, though the real issue I see here is that the new formatting is just a net negative. I think we would be better off to just make the source generator here emit an `#![allow(clippy::match_like_matches_macro)]`

**Comment by lnicola** [comment]
Well, some parts look better, others don't, and it's overall larger. But if it was written like this, we probably wouldn't have bothered to change it to `match`. So I'm neutral on this.

File: crates/syntax/src/ast/generated/nodes.rs [hunk 1]
```diff
@@ -3384,15 +3392,42 @@ impl From<UnderscoreExpr> for Expr {
...
-            | CAST_EXPR | CLOSURE_EXPR | CONTINUE_EXPR | FIELD_EXPR | FOR_EXPR | IF_EXPR
-            | INDEX_EXPR | LITERAL | LOOP_EXPR | MACRO_EXPR | MACRO_STMTS | MATCH_EXPR
-            | METHOD_CALL_EXPR | PAREN_EXPR | PATH_EXPR | PREFIX_EXPR | RANGE_EXPR
-            | RECORD_EXPR | REF_EXPR | RETURN_EXPR | TRY_EXPR | TUPLE_EXPR | WHILE_EXPR
-            | YIELD_EXPR | LET_EXPR | UNDERSCORE_EXPR => true,
-            _ => false,
-        }
+        matches!(
+            kind,
+            ARRAY_EXPR
```
Comment (lnicola) [inline crates/syntax/src/ast/generated/nodes.rs:3397]: These is actually a bit nicer when you're looking for one specific variant.

**Comment by Veykril** [comment]
Sure, I mean it really is me nitpicking at this point so it doesn't matter. One other benefit this formatting has is nicer diffs I suppose :)

@bors r+

**Pull request merged**